src/snitch_section.cpp

```diff
@@ -29,6 +38,18 @@ section_entry_checker::~section_entry_checker() {
             // that we don't know about yet. Popping will be done when we exit from the parent,
             // since then we will know if there is any sibling.
             state.sections.leaf_executed = true;
+#if SNITCH_WITH_TIMINGS
+            const auto  end_time = std::chrono::steady_clock::now();
+            const float duration = std::chrono::duration<float>(end_time - start_time).count();
+            state.reg.report_callback(
+                state.reg,
+                event::section_ended{
+                    data.id, data.location, false, asserts, failures, allowed_failures, duration});
+#else
+            state.reg.report_callback(
+                state.reg,
+                event::section_ended{data.id, data.location, asserts, failures, allowed_failures});
```

Comment:
I think all the reporting logic should be outsourced to `registry`; the `report_callback` isn't really meant to be used outside the `registry` class (it is exposed as a public member primarily so it can be assigned, otherwise it would be private). We probably want helper members functions like `registry::report_section_entered` / `registry::report_section_exited` to achieve this.

Reply:
This sounds much better. Adding the reporting here was the stuff I was most uncomfortable with and it seems very hacky.

Reply:
Done